src/EditorFeatures/Core/Tagging/IAsynchronousTaggerDataSource.cs

```diff
@@ -20,13 +20,21 @@ namespace Microsoft.CodeAnalysis.Editor.Tagging
     internal interface IAsynchronousTaggerDataSource<TTag> where TTag : ITag
```

Comment:
Do we really need this interface now? This would seem to make it possible for a consumer of this API to have their logic in a separate class than the provider, but the provider *also* implements this.